packages/sanity/src/_internal/cli/actions/schema/listSchemasAction.ts

```diff
@@ -1,33 +1,32 @@
 import {type CliCommandContext, type CliOutputter} from '@sanity/cli'
 import chalk from 'chalk'
 import sortBy from 'lodash/sortBy'
-import uniq from 'lodash/uniq'
 
 import {isDefined} from '../../../manifest/manifestTypeHelpers'
-import {
-  SANITY_WORKSPACE_SCHEMA_TYPE,
-  type StoredWorkspaceSchema,
-} from '../../../manifest/manifestTypes'
+import {type CreateManifest, type StoredWorkspaceSchema} from '../../../manifest/manifestTypes'
 import {type SchemaStoreActionResult, type SchemaStoreContext} from './schemaStoreTypes'
 import {createManifestExtractor, ensureManifestExtractSatisfied} from './utils/mainfestExtractor'
 import {createManifestReader} from './utils/manifestReader'
 import {createSchemaApiClient} from './utils/schemaApiClient'
-import {getDatasetsOutString} from './utils/schemaStoreOutStrings'
+import {getProjectIdDatasetsOutString, projectIdDatasetPair} from './utils/schemaStoreOutStrings'
 import {
-  filterLogReadProjectIdMismatch,
   parseListSchemasConfig,
+  SCHEMA_PERMISSION_HELP_TEXT,
   type SchemaStoreCommonFlags,
 } from './utils/schemaStoreValidation'
+import {uniqueProjectIdDataset} from './utils/uniqueProjectIdDataset'
 
 export interface SchemaListFlags extends SchemaStoreCommonFlags {
   json?: boolean
   id?: string
 }
 
 class DatasetError extends Error {
+  public projectId: string
   public dataset: string
-  constructor(dataset: string, options?: ErrorOptions) {
+  constructor(projectId: string, dataset: string, options?: ErrorOptions) {
```

Comment:
No blocker, but wonder if this should rather have been `constructor(message: string, info: {projectId: string, dataset:string), options?: ErrorOptions)`. I can easily see the list of things we want to include here grow over time.

Reply:
Heheh i thought about it when I added it, but still did it. Will fix.

Reply:
the one in delete is even worse, will fix that one too